apis/r/R/TileDBCreateOptions.R

```diff
@@ -0,0 +1,203 @@
+      o <- self$.dim(dim)
+      if (is.null(o)) {
+        return(default)
+      }
+      o <- o[['tile']]
+      if (is.null(o)) {
+        return(default)
+      }
+      return(o)
+    },
+
+    #' @return list of tiledb.Filter
+    offsets_filters = function(default=DEFAULT_OFFSETS_FILTERS()) {
+      self$.build_filters(super$get("offsets_filters", default))
+    },
+
+    #' @return list of tiledb.Filter
+    validity_filters = function(default=DEFAULT_VALIDITY_FILTERS()) {
+      self$.build_filters(super$get("validity_filters", default))
+    },
+
+    #' @return list of tiledb.Filter
+    dim_filters = function(default=DEFAULT_DIM_FILTERS()) {
+      self$.build_filters(super$get("dim_filters", default))
+    },
+
+    #' @return list of tiledb.Filter
+    attr_filters = function(default=DEFAULT_ATTR_FILTERS()) {
+      self$.build_filters(super$get("attr_filters", default))
+    },
+
+    # This is an accessor for nested things like
+    #
+    #   cfg <- PlatformConfig$new()
+    #   cfg$set('tiledb', 'create', 'dims', list(soma_dim_0 = list(tile = 6)))
+    #   tdco <- TileDBCreateOptions$new(cfg)
+    #
+    # -- this pulls out the `dims` -> `soma_dim_0` part.
+    #
+    #' return Named list of character
+    .dim = function(dim) {
+        o <- super$get("dims", NULL)
+        if (is.null(o)) {
+          return(NULL)
+        }
+        o[[dim]]
+    },
+
+    # This is an accessor for nested things like
+    #
+    #   cfg <- PlatformConfig$new()
+    #   cfg$set('tiledb', 'create', 'attrs', list(myattr = list(foo = 'bar')))
+    #   tdco <- TileDBCreateOptions$new(cfg)
+    #
+    # -- this pulls out the `attrs` -> `myattr` part.
+    #
+    #' return Named list of character
+    #' return Named list of character
+    .attr = function(attr) {
+        o <- super$get("attrs", NULL)
+        if (is.null(o)) {
+          return(NULL)
+        }
+        o[[attr]]
+    },
+
+    # The `items` argument is a list of arguments acceptable to `.build_filter`.
+    # Example argument:
+    #   list(
+    #     "RLE",
+    #     list(name = "ZSTD", COMPRESSION_LEVEL = 9)
+    #   )
+    # Returns a list of tiledb filters.
+    .build_filters = function(items) {
+      lapply(items, self$.build_filter)
+    },
+
+    # The `item` argument can be a string, like "RLE".
+    # Or, a named list with filter name and remaining arguments, like
+    # list(name="ZSTD", COMPRESSION_LEVEL=-1).
+    #
+    # Returns a tiledb filter.
+    .build_filter = function(item) {
+      # See also:
+      # https://tiledb-inc.github.io/TileDB-R/reference/tiledb_filter.html
+      if (is.character(item) && length(item) == 1) {
+        return(tiledb::tiledb_filter(item[[1]]))
+      }
+
+      stopifnot(!is.na(item['name']))
+      filter <- tiledb::tiledb_filter(item[['name']])
+      for (key in names(item)) {
+        cat("KEY", key, "\n")
+        cat("VAL", item[[key]], "\n")
+        cat("TYP", typeof(item[[key]]), "\n")
```

Comment:
If you make those `spdl::debug()` lines, possibly even a single one, there won't be a nag to remove it :wink:

```suggestion
spdl::debug("[build_filter] key {} val {} type {}", key, item[[key]], typeof(item[[key]]))
```

Reply:
no this is just wip debug on a draft pr with zero reviewers :)

thanks though!

Reply:
Well we all get a notification each time you commit so at times we do look ... :)